catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java

```diff
@@ -87,7 +87,14 @@ public void filter(ContainerRequestContext requestContext) {
     } catch (RuntimeException runtimeException) {
       throw new AuthenticationException("Invalid token");
     }
-    String authorizedEmail = jwt.getClaim("email").as(TextNode.class).asText();
+    String authorizedEmail;
+    if (jwt.getClaim("email") != null) {
+       authorizedEmail = jwt.getClaim("email").as(TextNode.class).asText();
+    } else if (jwt.getClaim("sub") != null){
+      authorizedEmail = jwt.getClaim("sub").as(TextNode.class).asText();
+    } else {
+      throw new AuthenticationException("Email id and subject not present");
+    }
     String userName = authorizedEmail.split("@")[0];
```

Comment:
Wouldn't this throw an error for the subject part as it won't be able to split "@"

Reply:
In subject, the value we get is the client_id@clients i.e.  `"sub": "nLoFA7ZA8xxxxxxxxxxxxxxJpk3@clients"`. So it won't throw any error.

Reply:
Is that consistent across all the SSOs we support

Reply:
No, will look into it.